modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/PageRenderer/Layout.es.js

```diff
@@ -29,6 +31,35 @@ export function Layout({components, editable, itemPath, rows, viewMode}) {
 
 	const Components = components ?? mergeVariants(editable, variants);
 
+	useEffect(() => {
+		const undoHandler = () => {
+			dispatch({type: EVENT_TYPES.HISTORY.PREV});
+		};
+		const redoHandler = () => {
+			dispatch({type: EVENT_TYPES.HISTORY.NEXT});
+		};
+
+		Liferay.on('undo', undoHandler);
+		Liferay.on('redo', redoHandler);
+
+		return () => {
+			Liferay.detach('undo', undoHandler);
+			Liferay.detach('redo', redoHandler);
```

Comment:
It's probably not a great idea for a generic "undo" and "redo" event here. Can you namespace those events to your module? Otherwise other devs may we undo/redo events pop up and assume they can hook into it.